File: network/custom-chaincode/chaincode/lib/model/PatientResource.js
'use strict';

const Resource = require('./Resource.js');

class PatientResource extends Resource {
    constructor(resourceType, id, gender, telecom, name, birthDate, managingOrganization, generalPractitioner) {
        super(resourceType, id, gender, telecom, name, birthDate, managingOrganization);
    }
}

module.exports = PatientResource;

File: backend/src/main/java/com/healthchain/backend/model/resource/PatientResource.java
package com.healthchain.backend.model.resource;

import lombok.Data;

import java.util.List;

@Data
public class PatientResource extends Resource{
    private List<Reference> generalPractitioner;
}

//{
//        "resourceType": "Patient", //Patient / Practitioner
//        "id": "123456",
//        "gender": "male",
//        "telecom": [
//        {
//        "system": "phone",
//        "value": "555-1234"
//        },
//        {
//        "system": "email",
//        "value": "example@email.com"
//        }
//        ],
//        "name": [
//        {
//        "use": "official",
//        "family": "Doe",
//        "given": ["John", "M."]
//        }
//        ],
//        "birthDate": "1980-01-01"
//        }

